### PR TITLE
Added lightcone integration of supernovae density

### PR DESCRIPTION
**Pull request by cmistick**
- Added SNeLightcone class in `supernovae_lightcone.py`
  - Utilizes skypy function `redshifts_from_comoving_density` to obtain an array of supernovae redshift locations from SNe comoving density
  - Allows user to pick the time interval and redshift range over which the lightcone is evaluated
  - Can output the redshifts as an astropy table
- Added corresponding TestSNeLightcone class in `test_supernovae_lightcone.py`
- Updated `supernovae_pop.py` to allow array-like inputs to facilitate SNeLightcone class 
- Updated corresponding `test_supernovae_pop.py`

### CODE REVIEW

File: slsim/Sources/Supernovae/supernovae_lightcone.py
Comment (nkhadka21) [inline slsim/Sources/Supernovae/supernovae_lightcone.py:46]: Might be good rename it as supernovae_sample().

File: slsim/Sources/Supernovae/supernovae_lightcone.py
Comment (nkhadka21) [inline slsim/Sources/Supernovae/supernovae_lightcone.py:62]: Instead of making redshift_list an input, you can directly call return_supernovae_sample() function inside this function.

**Review by nkhadka21 — CHANGES_REQUESTED** [review]
Hi @cmistick, thank you very much for this PR! It looks very good. I have minor comments. Please look at them.

File: slsim/Sources/Supernovae/supernovae_lightcone.py
Comment (cmistick) [reply slsim/Sources/Supernovae/supernovae_lightcone.py:46]: I didn't want to name it the same thing as supernovae_sample.py, but if it won't cause any problems then I'll change it.

File: slsim/Sources/Supernovae/supernovae_lightcone.py
Comment (sibirrer) [inline slsim/Sources/Supernovae/supernovae_lightcone.py:34]: somewhere the little h got missed, or where do you take it into account?

File: slsim/Sources/Supernovae/supernovae_lightcone.py
Comment (sibirrer) [inline slsim/Sources/Supernovae/supernovae_lightcone.py:69]: could it be that if you execute this routine twice that the same class would return you different supernovae? It might be better to generate the supernovae_sample only once and then re-use it within the same class instance

File: slsim/Sources/Supernovae/supernovae_pop.py
Comment (sibirrer) [inline slsim/Sources/Supernovae/supernovae_pop.py:24]: what is the unit of the delay time?

**Review by sibirrer — CHANGES_REQUESTED** [review]
Looks great @cmistick. I have few minor comments

File: slsim/Sources/Supernovae/supernovae_lightcone.py
Comment (cmistick) [reply slsim/Sources/Supernovae/supernovae_lightcone.py:34]: For some reason I keep forgetting to put it in the units and then forgot to divide by the factor. I divided by the factor in the initialization function when `self.density` is calculated.

**Review by nkhadka21 — APPROVED** [review]
Hi @cmistick, thank you very much for this update! It looks great. I approve and wait for @sibirrer approval.

**Review by sibirrer — APPROVED** [review]
Looks great!

**Pull request merged**
